Fix skflow resnet example to work on v0.10.0

tyfkda opened the pull request:
I tried to run `resnet.py` skflow example on TensorFlow v0.10.0,
but an error occurred.

I have fixed the code to run on v0.10.0:
- `tensorflow.contrib.learn.ops.conv2d` was deprecated and has been removed in #4373 .
  So changed to use `contrib.layers.convolution2d` .
- `classifier` wasn't created if no previous model existed in `models/resnet/graph.pbtxt`,
  but it was failed to run at first time.
  `TensorFlowEstimator` creation was removed in #2863 , so revert the code removal.

tyfkda committed:
Fix skflow resnet example to work on v0.10.0

`tensorflow.contrib.learn.ops.conv2d` was deprecated and has been removed in #4373 .
So changed to use `contrib.layers.convolution2d` .

`classifier` wasn't created if no previous model existed in `models/resnet/graph.pbtxt`,
but it was failed to run at first time.
`TensorFlowEstimator` creation was removed in #2863 ,
so revert the code removal.

`TensorFlowEstimator#restore` raises `NotImplementedError`,
so remove model restoration.

tensorflow-jenkins commented:
Can one of the admins verify this patch?

terrytangyuan commented:
Could you change to use Estimator and make sure the changes work with the version on master branch?

tyfkda commented:
@terrytangyuan ,
Can I make your suggestion in other pull request to keep this change simple?

I am trying to modify the code to use `Estimator` instead of `TesorFlowEstimator`, but it looks the class lacks functionality. Especially I cannot pass `feed_fn` to it, which is used in [TensorFlowEstimator#fit](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/learn/python/learn/estimators/base.py#L254).

I have sent a pull request to add the functionality (#4696). When it is accepted, I will change this example code to use `Estimator`.

terrytangyuan commented:
Sure, perhaps could you follow other examples that are being tested (you can take a look at the shell script in examples folder) so this example is being tested going forward? You might need to make sure it doesn't take forever to run, using the smaller dataset provided in TF.Learn.datasets.

tyfkda commented:
@terrytangyuan 
I've added a commit which adds resnet.py to the shell script.
I use `100` for test count, I am not sure it is good for CI.

terrytangyuan commented:
Thanks @tensorflow-jenkins Test this please

tyfkda commented:
Test failed, I should add the target into `BUILD`.
I've fixed the commit.

terrytangyuan commented:
@tensorflow-jenkins test this please

tyfkda commented:
@terrytangyuan Would you please take a look?
Test failed because the code used `tensorflow.examples.tutorials` package.

I have changed the code to use `learn.datasets.load_dataset`,
referred mnist.py example.

gideonite commented on tensorflow/examples/skflow/resnet.py:
```diff
@@ -141,13 +135,14 @@ def res_net(x, y, activation=tf.nn.relu):
+    learning_rate=0.001, continue_training=True)
 
-while True:
+for step in xrange(100):
```
nit pick: I can see why you did this but it's besides the point. Not important either way since this is just an example after all.

tyfkda replied:
This change avoids infinite loop for CI test.

gideonite reviewed (approved):
I'm not familiar with the new skflow API but LGTM.

yifeif commented:
Looks like there are some conflicts. @tyfkda could you resolve them? Thanks!

tyfkda committed:
Add resnet example to test

To avoid breakage, add resnet.py to test.

tyfkda commented:
@yifeif Thank you for the notice, I have fixed the conflict, would you please take a look?

It seems examples have moved from tensorflow/examples/skflow to tensorflow/examples/learn in d74cc130dd0ce5ecd1421ef126e4446f337a856c (#3364),
so I moved resnet.py example accordingly.

martinwicke commented on tensorflow/examples/learn/resnet.py:
```diff
@@ -141,13 +135,14 @@ def res_net(x, y, activation=tf.nn.relu):
-if os.path.exists('models/resnet/graph.pbtxt'):
-  classifier = learn.TensorFlowEstimator.restore('models/resnet/')
+# Create a new resnet classifier.
+classifier = learn.TensorFlowEstimator(
```
We are trying to get rid of the (deprecated) TensorFlowEstimator. Can you rewrite this as an Estimator instead? It's a fairly straightforward translation -- include the optimize_loss call in your model_fn, which has to return also a train_op (which the optimize_loss call creates for you).

tyfkda replied:
OK, I have updated the code to use `Estimator`.

martinwicke commented on tensorflow/examples/learn/resnet.py:
```diff
@@ -141,13 +135,14 @@ def res_net(x, y, activation=tf.nn.relu):
+    learning_rate=0.001, continue_training=True)
 
-while True:
+for step in xrange(100):
```
fit() should never be called in a loop. I know this is from an old example and you didn't do it, but as part of moving this example to the learn/ directory we have to update it to current best practices.

Instead, use steps=100, and if you want to output a score regularly, use a ValidationMonitor.

tyfkda committed:
Use Estimator instead of TensorFlowEstimator in resnet.py example

tyfkda commented:
@martinwicke Would you please take a look? Thanks!

martinwicke commented on tensorflow/examples/learn/resnet.py:
```diff
@@ -138,21 +132,31 @@ def res_net(x, y, activation=tf.nn.relu):
-# Restore model if graph is saved into a folder.
-if os.path.exists('models/resnet/graph.pbtxt'):
-  classifier = learn.TensorFlowEstimator.restore('models/resnet/')
+tf.logging.set_verbosity(tf.logging.INFO)  # Show training logs. (avoid sirence)
```
sirence => silence

tyfkda replied:
Fixed.

martinwicke commented on tensorflow/examples/learn/resnet.py:
```diff
@@ -138,21 +132,31 @@ def res_net(x, y, activation=tf.nn.relu):
-  score = metrics.accuracy_score(
-      mnist.test.labels, classifier.predict(mnist.test.images, batch_size=64))
-  print('Accuracy: {0:f}'.format(score))
+# Calculate accuracy.
```
Can you use `classifier.evaluate(... metrics=[streaming_accuracy])`?

tyfkda replied:
I have updated the code to use `classifer.evaluate`.

martinwicke reviewed (changes requested):
Looks good -- two small changes? Sorry it took me so long.

tyfkda commented:
@martinwicke Would you please take a look? Thanks!

martinwicke commented:
Jenkins, test this please!

martinwicke commented:
Looks great, thanks for doing this!

tyfkda commented:
I can use `Estimator#fit` with passing `x` and `y` argument,
so solved the problem.

I am going to close this issue.
